managed/services/backup/backup_service.go

```diff
@@ -288,14 +288,17 @@ func (s *Service) RestoreBackup(ctx context.Context, serviceID, artifactID strin
 			jobType = models.MySQLRestoreBackupJob
 			jobData = &models.JobData{
 				MySQLRestoreBackup: &models.MySQLRestoreBackupJobData{
+					ServiceID: serviceID,
 					RestoreID: restoreID,
 				},
 			}
 		case models.MongoDBServiceType:
 			jobType = models.MongoDBRestoreBackupJob
 			jobData = &models.JobData{
 				MongoDBRestoreBackup: &models.MongoDBRestoreBackupJobData{
+					ServiceID: serviceID,
 					RestoreID: restoreID,
+					DataModel: params.DataModel,
```

Comment:
We have this information in artifact. Do we really need it here?

Reply:
I added it to make sure we only restart for physical backups. The alternative will be to use restore ID to get artifact ID, then use artifact ID to get the data model - and this felt simpler.